### DISCUSSION TITLE
Future file format

### DISCUSSION THREAD

**Discussion by micahellison** [jrnl-org/jrnl#1610] [Ideas]
@wren and I have been talking about a new file format for jrnl a lot, and we want to open up the discussion to users and contributors.

I think the capstone issue around this is the very reasonable desire for a unique identifier for each entry (#519). There's also #1609 to allow sorting within the storage format.

We know this will be a breaking change to jrnl, with lots of downstream implications, so we will be approaching this change very carefully. However, whatever happens, we do intend to keep the existing format as an option.

Feel free to add your thoughts and link issues.

**Comment by wren** [comment]
Here are some issues/feature requests that have come up that we should consider when designing the new format (not necessarily guarantee that the new format will support everything, just that we should consider whether we want to or not):

- #1390
- #1380
- #1026
- #1149
- #1289
- #1098
- #949
- #696
- #677
- #1051
- #779
- #588
- #444

**Comment by micahellison** [comment]
A couple more features we're thinking would be nice in the storage format:

* Storing start and end times for an entry. Maybe even automatically -- jrnl could record the start time when composing a new entry then save the end time when finished.
* "All day" times - basically dates with no time components. Very common in calendaring applications and could be useful here. Thanks @DSiekmeier for the idea in #1465

**Reply by micahellison** [reply]
Re: start and end times - see issue #1637

**Comment by micahellison** [comment]
See also #1683 - Make it possible to add images
